web-ng/src/app/components/form-field-editor/form-field-editor.component.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Component,
+  OnInit,
+  Input,
+  Output,
+  EventEmitter,
+  OnChanges,
+} from '@angular/core';
+import { FormGroup, FormBuilder } from '@angular/forms';
+import { FieldType } from '../../shared/models/form/field.model';
+import { StringMap } from '../../shared/models/string-map.model';
+
+export interface FieldTypeModel {
+  icon: string;
+  label: string;
+  type: FieldType;
+}
+
+@Component({
+  selector: 'app-form-field-editor',
+  templateUrl: './form-field-editor.component.html',
+  styleUrls: ['./form-field-editor.component.css'],
+})
+export class FormFieldEditorComponent implements OnInit, OnChanges {
+  @Input() label?: string;
+  @Input() required?: boolean;
+  @Input() type?: string;
+  @Output() update = new EventEmitter();
+  @Output() delete = new EventEmitter();
+  fieldTypes: FieldTypeModel[] = [
```

Comment:
If we won't move these into the template, let's mark the field as readonly to indicate it will never be replaced. 

Also, as mentioned "Model" may be confusing here; `FieldType` may be clearer.